src/Microsoft.ML.Data/Data/SchemaDefinition.cs

```diff
@@ -17,50 +17,65 @@ namespace Microsoft.ML.Data
     [AttributeUsage(AttributeTargets.Field | AttributeTargets.Property, AllowMultiple = false, Inherited = true)]
     public sealed class KeyTypeAttribute : Attribute
     {
-        // REVIEW: Property based, but should I just have a constructor?
+        /// <summary>
+        /// Specifies expected size of key type.
+        /// </summary>
+        /// <param name="count">Size of key type.</param>
+        public KeyTypeAttribute(ulong count)
+        {
+            Count = count;
+        }
 
         /// <summary>
         /// The key count.
         /// </summary>
-        public ulong Count { get; set; }
+        internal ulong Count { get; }
     }
 
     /// <summary>
     /// Allows a member to be marked as a vector valued field, primarily allowing one to set
     /// the dimensionality of the resulting array.
     /// </summary>
+    /// <remarks>
+    /// For vector with variable size use <see cref="VariableVectorTypeAttribute"/>.
+    /// </remarks>
     [AttributeUsage(AttributeTargets.Field | AttributeTargets.Property, AllowMultiple = false, Inherited = true)]
     public sealed class VectorTypeAttribute : Attribute
     {
-        private readonly int[] _dims;
-
         /// <summary>
         /// The length of the vectors from this vector valued field.
         /// </summary>
-        public int[] Dims { get { return _dims; } }
+        internal int[] Dims { get; }
+
+        /// <summary>
+        /// Mark member with expected size of array.
+        /// </summary>
+        /// <param name="size">Expected size of array.</param>
+        public VectorTypeAttribute(int size)
+        {
+            Contracts.CheckParam(size > 0, nameof(size), "Should be positive number, if you want to specify variable size please use " + nameof(VariableVectorTypeAttribute));
+            Dims = new int[1] { size };
+        }
 
         public VectorTypeAttribute(params int[] dims)
```

Comment:
>dims [](start = 48, length = 4)

So long as you're cleaning up the public surface, even though it's a parameter name we should probably still follow .NET naming guidelines and avoid abbreviations and so on.

Reply:
oops

---
In reply to: [258276176](https://github.com/dotnet/machinelearning/pull/2638#discussion_r258276176) [](ancestors = 258276176)